### ISSUE TITLE
According to ISO 8601, the month starts with 1 through 12

### ISSUE THREAD

**Issue by meness** [iamkun/dayjs#614]
Hi everyone,

Please follow standards here: https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates.

According to ISO 8601, the month starts with 1 through 12. Most of the libraries (i.e., NgBootstrap) follow ISO 8601 and this produces bugs while using dayjs.

Thanks!

**Comment by iamkun** [comment]
We followed Javascript Date Object that starts at 0 at the moment.

**Comment by meness** [comment]
> We followed Javascript Date Object that starts at 0 at the moment.

You are right, the current implementation of JS Date uses 0-index for the month. Is it possible to provide a boolean for `dayjs()` to make it follow ISO 8601?

**Comment by iamkun** [comment]
What do you mean "provide a boolean for dayjs() ", pls?

**Comment by meness** [comment]
> What do you mean "provide a boolean for dayjs() ", pls?

As there are two different implementations for the month, 0-index and 1-index, it is good to give developers a way to force `Dayjs` objects follow one of the implementations.

```typescript
// Global options
export interface DayjsOptions {
  followIso8601: boolean // Default: false
}

Dayjs.options = {followIso8601: true}

dayjs().month() // Returns 1-index month number.
```

**Comment by hakimio** [comment]
@meness Any reason you can't just call format() or simply add 1 yourself? Seems like an over-complication with the config.

**Comment by meness** [comment]
> @meness Any reason you can't just call format() or simply add 1 yourself? Seems like an over-complication with the config.

It is not really an issue right now but I like to see it later because of easy to use API point of view.

**Comment by ypresto** [comment]
I produced many bugs caused by forget to `month = jsMonth + 1` or `jsMonth = month - 1`...
IMO JavaScript specification is the "justice", but I also agree that we need some fixing for this `(month - 1)` problem. :joy:

**Comment by ypresto** [comment]
Maybe something like `dayjs().monthOfYear()` or `dayjs().naturalMonth()`, and `dayjs.natural(2019, 1, 23)`..?
(BTW we already have Date.getFullYear() to fix original JS specification 🤔)

**Comment by iamkun** [comment]
@neemzy Well, this is a question with no right or wrong answer.  We followed the Javascript Date Object that starts at 0 and may never change.

However, we could provide a plugin or config to make it starts at 1 if there's a need. I'm more worried about it makes more confusion.

**Comment by wmonecke** [comment]
Hi! Any update on this? Is there an official plugin?

**Comment by floinay** [comment]
any update of this?
